src/main/java/hudson/plugins/xshell/XShellBuilder.java

```diff
@@ -115,9 +125,12 @@ public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher,
     EnvVars env = build.getEnvironment(listener);
     env.putAll(build.getBuildVariables());
 
+    // Determine the correct working directory
+    String absWorkingDir = build.getWorkspace() + (launcher.isUnix() ? UNIX_SEP : WINDOWS_SEP) + workingDir;

```

Comment:
The working dir has to be relative to the workspace.
Is there a way to allow an absolute path?